x-pack/plugins/security_solution/server/fleet_integration/fleet_integration.ts

```diff
@@ -104,6 +105,7 @@ export const getPackagePolicyCreateCallback = (
     }
 
     if (newPackagePolicy?.inputs) {
+      validatePolicyAgainstAppFeatures(newPackagePolicy.inputs, appFeatures);
```

Comment:
This validation seems to be in a Kibana request context. 

If that's the case, when we have RBAC for the `endpointPolicyProtection` feature (not done yet in this PR but it will need to be done eventually), some users won't be allowed to do this action even if the appFeature is enabled. So, we will need to validate this feature against the current user privileges, instead of checking the AppFeatures service directly. Otherwise, we will introduce an authorization bug, allowing the action to unauthorized users.

(consider configuring the `privileges` for this appFeature in this PR, and do the user privilege validation. Otherwise, make sure we remember to change all the appFeature validations inside API requests later on)

Reply:
Thanks for pointing this out @semd !
@paul-tavares @ashokaditya Do you think both appFeature and user privilege check should happen here, in this place?

Reply:
If we do the user privilege check we don't need to check the appFeature, that's implicit. 
However, I see other raw AppFeatures checks in this same file, so I don't know whether the user privileges are relevant in this logic or not, I am guessing they are since we have the `request` object here.

When I say "check the user privileges" I mean adding a new Kibana privilege using AppFeature config, that gets integrated into the "endpointAuthz" RBAC with a new key `canUseEndpointProtection`. Like the other endpoint features that were implemented previously.

Reply:
In this case, the feature we are trying to gate follows permissions of `canReadPolicyManagement` in UI (which utilizes https://github.com/elastic/kibana/blob/a8de031ddf9b9d172a4901db85341930702cb1c8/x-pack/plugins/security_solution/common/endpoint/service/authz/authz.ts#L48) . Unless we want to define an explicit key for `protectionUpdates` we can reuse it. However, I don't see connection between `appFeatures` and `calculateEndpointAuthz`, maybe this is something that could be refactored into using Kibana privilege?

Reply:
> Unless we want to define an explicit key for protectionUpdates 

As per the serverless predefined roles definition, a new privilege for `Agent / Endpoint Tamper protection` needs to be created and assigned to different roles:

![role_privilege](https://github.com/elastic/kibana/assets/17747913/426722ed-127d-4140-b903-055085a0b00b)

I am not aware of all the parts of the application that will need to check this feature privilege

> I don't see connection between appFeatures and calculateEndpointAuthz

The connection is not obvious:
The `AppFeatures` service registers the `Kibana feature` configuration for Security Solution app, including all Endpoint sub-features, `calculateEndpointAuthz` checks those Kibana features to generate all the authz keys.

`Security AppFeatures` --(registers)--> `Kibana features` --(are checked by)--> `calculateEndpointAuthz`

Reply:
I think you meant this `canWritePolicyManagement` instead of `Agent / Endpoint Tamper protection`?
If this is the case then we will need to update existing roles @dasansol92 and introduce this new privilege. 

To sum up, once we introduce this `canWritePolicyManagement` privilege and assign it to different serverless roles we should remove this request appFeature check in favor of user privilege check that will do both - check user privileges and tied appFeature. Is this correct @semd ?

Reply:
Agent tamper protection is other feature not related to this one. 
I guess this is tied to `canWritePolicyManagement` since we are doing integration policy updates here, but for serverless we might want to add new permission and add it to a role. 
This permission should be checked for manual actions only I guess. On the package policy watcher context, we should rely on appFeatures (or licensing for ESS) to turn them off (set it to `latest`) if there is a downgrade. Is this right @caitlinbetz ?

Reply:
@dasansol92 Agree, they are two separate features in the spreadsheet, with different role assignations as well.
![role_features](https://github.com/elastic/kibana/assets/17747913/e9aa4d67-6f89-476c-a391-ce471040f4b2)

Reply:
Resolving since we won't be pursuing separate RBAC for this feature.